scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHandshaker.java

```diff
@@ -120,8 +121,10 @@
 	protected final ServerNameResolver serverNameResolver;
 	protected ServerNames indicatedServerNames;
 	protected SignatureAndHashAlgorithm negotiatedSignatureAndHashAlgorithm;
-
-
+ 
+    /** The trusted raw public keys */
+    protected TrustedRpkStore rpkStore;
```

Comment:
If that is moved to the base class `Handshaker`, `verifyCertificate` would only require the `CertificateMessage` as parameter (the `rootCertificates` are already there).

Reply:
And consider to use final.